theia-extensions/cdt-cloud-pico/src/browser/project-editor/project-editor-widget.tsx

```diff
@@ -0,0 +1,226 @@
+                    <div className="setting-item-header">
+                        <h3>
+                            Binary URL
+                        </h3>
+                    </div>
+                    <div className="setting-item-description">
+                        <p>Controls the <i>project URL</i> written into the binary.</p>
+                    </div>
+                    <div>
+                        <div className='theia-select-component'>
+                            <input
+                                className='theia-input'
+                                type='text'
+                                value={this.projectSettings.url ?? ''}
+                                onChange={e => this.setProjectSettingsUrl(e)} />
+                        </div>
+                    </div>
+                </div>
+                <div className="setting-item-contents settings-row-inner-container">
+                    <div className="setting-item-header">
+                        <h3>
+                            Serial Printing
+                        </h3>
+                    </div>
+                    <div className="setting-item-description">
+                        <p>Controls the output channel of the serial printing.</p>
+                    </div>
+                    <div>
+                        <div className='theia-select-component'>
+                            <select onChange={e => this.setProjectSettingsSerialPrinting(e)}>
+                                <option value='USB' selected={this.projectSettings.serialPrinting === 'USB'}>USB</option>
+                                <option value='UART' selected={this.projectSettings.serialPrinting === 'UART'}>UART</option>
+                                <option value='Both' selected={this.projectSettings.serialPrinting === 'Both'}>Both</option>
+                                <option value='None' selected={this.projectSettings.serialPrinting === 'None'}>None</option>
+                            </select>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>;
+    }
+
+    protected setProjectSettingsName(e: React.ChangeEvent<HTMLInputElement>): void {
+        this.projectSettings!.name = e.target.value;
+        this.saveProjectSettings();
+    }
+
+    protected setProjectSettingsUrl(e: React.ChangeEvent<HTMLInputElement>): void {
+        this.projectSettings!.url = e.target.value;
+        this.saveProjectSettings();
+    }
+
+    protected setProjectSettingsSerialPrinting(e: React.ChangeEvent<HTMLSelectElement>): void {
+        this.projectSettings!.serialPrinting = e.target.value;
+        this.saveProjectSettings();
+    }
+
+    protected async saveProjectSettings(): Promise<void> {
+        this.fileService.writeFile(
+            this.projectSettingsUri,
+            BinaryBuffer.fromString(JSON.stringify(this.projectSettings))
```

Comment:
Minor: Beautify JSON string on save
```suggestion
            BinaryBuffer.fromString(JSON.stringify(this.projectSettings, undefined, 2))
```